### PR TITLE
Add onlyNew flag to PubSub and PubSubAsyncIterator which allow sending only new messages to the subscribed client

### PR DESCRIPTION
**Pull request by vrehor**

### CODE REVIEW

**Comment by swantzter** [comment]
Nice, thank you.

**Comment by swantzter** [comment]
Hmm the test seem to be failing because node's timer seem to be drifting in comparison with the firebase emulator's timer.

Do you want to maybe change https://github.com/swantzter/graphql-firebase-subscriptions/pull/4/files#diff-cb731e23a3f256d9c23da3ac34b798a3262046bd0feff41c44d49f527c307d88R51 to use `Date.now()` instead of `ServerValue.TIEMSTAMP`

File: src/async-iterator.ts [hunk 1]
```diff
@@ -2,18 +2,21 @@ import { PubSubEngine } from 'graphql-subscriptions'
 import { $$asyncIterator } from 'iterall'
 
 export class PubSubAsyncIterator<T> implements AsyncIterator<T> {
+  private readonly createdAt = new Date().getTime()
```
Comment (swantzter) [inline src/async-iterator.ts]:
```suggestion
  private readonly createdAt = Date.now()
```

I think you could just use this

**Review by swantzter — COMMENTED** [review]
really small nitpick, but if you're changing `ServerValue.TIMESTAMP` already I kinda prefer using `Date.now()` here

File: src/async-iterator.ts (hunk 1 above)
Comment (vrehor) [reply src/async-iterator.ts]: I can try. But still there is problem, when your backend environment has wrong time. The only one solution which can fix this is to get date from firebase. But there is no simple solution to get it.

How to do it:
1. create and save object {timestamp:ServerValue.TIMESTAMP}
2. read the timestamp value
3. save it to createAt

But this is overkill. What you think?

Comment (vrehor) [reply src/async-iterator.ts]: Or to use Date.now() instead of ServerValue.TIMESTAMP everywhere

Comment (swantzter) [reply src/async-iterator.ts]: @vrehor yeah I think let's just swap out `ServerValue.TIMESTAMP` for `Date.now()` on L51 in `pubsub.ts` like you suggested - That should probably give as good of a time value anyways

Alternatively you can allow a bit of grace period, so `createdAt = Date.now() - 1000` or something like that in L5 of async-iterator.ts?

I think either of those solutions would be better than trying to read back the value of ServerValue.TIMESTAMP, feel free to take whichever you prefer :)

Comment (vrehor) [reply src/async-iterator.ts]: @swantzter I picked the first one. Now it is synchronized with getDeletionRoutineFunction where Date.now function is also used :-)

**Review by swantzter — APPROVED** [review]

**Pull request merged**